### PR TITLE
[Location sharing] Invisible text on map symbol

### LINKED ISSUE #6687: [Location sharing] Invisible text on map symbol
### Steps to reproduce

- Share you current location in a room with a user without picture
- Open the new message

#### What did you expect?
The map pin should display the shortname of the user on user color background.

#### What happened instead?
The map pin displays only the background color.

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

_No response_

### Homeserver

_No response_

### Will you send logs?

No

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Convert the map symbol drawable to bitmap to be able to see text on it.

## Motivation and context

Closes #6687 

## Screenshots / GIFs

|Before|After|
|-|-|
|||

## Tests

- Share you current location in a room with a user without picture
- Open the new message
- Check the shortname of the user is displayed on the map pin

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by ouchadam** [comment]
it would be handy to have screenshots for this fix :pray:

File: vector/src/main/java/im/vector/app/features/location/MapTilerMapView.kt [hunk 1]
```diff
@@ -162,7 +163,7 @@ class MapTilerMapView @JvmOverloads constructor(
         pinDrawable?.let { drawable ->
             if (!safeMapRefs.style.isFullyLoaded ||
                     safeMapRefs.style.getImage(state.pinId) == null) {
-                safeMapRefs.style.addImage(state.pinId, drawable)
+                safeMapRefs.style.addImage(state.pinId, drawable.toBitmap())
```
Comment (ouchadam) [inline vector/src/main/java/im/vector/app/features/location/MapTilerMapView.kt:166]: do we know why this behaviour is different? 

`addImage(drawable)` chains into `BitmapUtils.getBitmapFromDrawable(drawable)`  which in turn calls `Bitmap.createBitmap(drawable.getIntrinsicWidth(), drawable.getIntrinsicHeight(), Bitmap.Config.ARGB_8888)`

whereas `drawable.toBitmap()` calls `Bitmap.createBitmap(width, height, config ?: Config.ARGB_8888)`

I would have thought them to be equivalent? :thinking:

Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/MapTilerMapView.kt:166]: I wasn't curious enough to understand... Let me look at it to answer this question.

Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/MapTilerMapView.kt:166]: After some investigation it seems the only difference I see is that in `BitmapUtils.getBitmapFromDrawable()` they are creating a new drawable with `Drawable drawable = constantState.newDrawable().mutate();` and then the bitmap is created from this new drawable. I suspect the Text part disappears due to this call. In the doc, it says density dependent properties may not be scaled well (https://developer.android.com/reference/android/graphics/drawable/Drawable.ConstantState#newDrawable()). Testing with following code leads to the same issue:
```
val bitmap = drawable.constantState?.newDrawable()?.toBitmap()
bitmap?.let { safeMapRefs.style.addImage(state.pinId, it) }
```

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/location/MapTilerMapView.kt:166]: great to know, thanks for looking into it! :100:

**Review by ouchadam — APPROVED** [review]

**Pull request merged**